### PR TITLE
Added elasticache-cluster terraform module ref: saas-11511

### PR DESCRIPTION
**Pull request by rameshganne**
Reference to the ticket id [SAAS-11511](https://dimagi-dev.atlassian.net/browse/SAAS-11511), I am adding terraform modules (elasticache-cluster, r53-private-zone-create-update & r53-record-create-update ). 

Information about modules:
elasticache-cluster: It can create AWS elasticache service with cluster mode. 
r53-private-zone-create-update: It can create Route-53 private hosted-zone and records.
r53-record-create-update: It can create new records if the Route-53 hosted-zone already exist.

Extend to the above modules, I am updating terraform template files, environment and environment schema file to get terraform variable values from environment based files by using code.

**src/commcare_cloud/commands/terraform/templates/commcarehq.tf.j2**  -- In this template file, provided terraform modules (elasticache-cluster & r53-private-zone-create-update) in jinja template format to create commcarehq.tf file with required data to execute elasticache-cluster and r53-private-zone-create-update module
**environments/staging/terraform.yml** -- In this environment file, provided required variable values for staging environment.
**src/commcare_cloud/environment/schemas/terraform.py** -- In this schema file, added code to create terraform file by reading template file & environment file. 

Here we can create AWS Elasticache cluster, Route-53 hosted zone and add record-set with cluster end-point. These changes can apply for staging environment only. At this level, we are not adding integration part changes(application & redis [elasticache endpoint] ).

### CODE REVIEW

File: src/commcare_cloud/terraform/modules/r53-private-zone-create-update/main.tf-backup [hunk 1]
```diff
@@ -0,0 +1,14 @@
+resource "aws_route53_zone" "elastic-cache-dns" {
```
Comment (dannyroberts) [inline src/commcare_cloud/terraform/modules/r53-private-zone-create-update/main.tf-backup:1]: This file is called `main.tf-backup` which makes me think this wasn't intended to be committed. Is that right?

**Review by dannyroberts — APPROVED** [review]
looks good but for my one comment. @sanjay2916 can you check that this also runs cleanly on india and production without further changes to their configuration?

File: src/commcare_cloud/terraform/modules/r53-private-zone-create-update/main.tf-backup (hunk 1 above)
Comment (rameshganne) [reply src/commcare_cloud/terraform/modules/r53-private-zone-create-update/main.tf-backup]: Yes @dannyroberts. I have deleted it.

**Comment by rameshganne** [comment]
> looks good but for my one comment. @sanjay2916 can you check that this also runs cleanly on india and production without further changes to their configuration?

Presently, I made changes for staging environment only. For India & Production environment case, we will be changing below environment files with terraform variables & values.
---
environments/india/terraform.yml
environments/production/terraform.yml
---

**Comment by sanjay2916** [comment]
> > looks good but for my one comment. @sanjay2916 can you check that this also runs cleanly on india and production without further changes to their configuration?
> 
> Presently, I made changes for staging environment only. For India & Production environment case, we will be changing below environment files with terraform variables & values.
> 
> * environments/india/terraform.yml
> * environments/production/terraform.yml

@rameshganne I will test this India and Production environment

**Comment by dannyroberts** [comment]
I didn't mean to leave you guys hanging here. Not sure if you already tested, but just let me know and I can merge this

**Comment by sanjay2916** [comment]
@rameshganne @shyamkumarlchauhan  I have tested in staging and able to crate the Elaticache Cluster

**Pull request merged**